_posts/2025-08-26-openloco-v25.08.md

```diff
@@ -0,0 +1,133 @@
+multiply, die, and be covered in snow. There are a huge amount of things that can trigger the tree
+death! And tree death is a whole animation in itself; it's not just a simple case of removing the
+tree.
+
+## Vehicle Route Finding
+
+With the AI now finished, it was time to move focus to the remaining vehicle functions. The majority
+of the remaining vehicle code all had interactions with routing. I had purposely left these
+functions until the end to ensure we understood how things worked. I'm still not 100% sure, but I
+think I get the gist of it.
+
+All vehicles are comprised of a number of cars and a number of hidden components. All of these 
+components have their own positions and are updated separately. They all share what I have called
+a routing. A routing is a list of track ids that the vehicle will follow. The maximum length of the
+routing is 64 track ids. (This is why you can't place a train that takes up more than 64 track
+pieces.) It's not quite as simple as 'head of train is the first routing entry', though. There is an
+invisible component that runs in front of the head of the train (I think) and there is an invisible
+component trailing at the back of the train (I think).
+
+Whenever a vehicle comes up to a junction, it has to decide on a routing. This is the route finding
+code. It works mostly how you would expect, i.e. find the shortest path to the destination. It can
+only look ahead a certain distance, and only a max of 5 functions. It also has to take into account
+signals, which will hopefully indicate if there is a train blocking a route. Track and road route
+finding look to be mostly the same, but with the additional complexity of signals for track and
+multiple lanes for road. There are three variations of the route finding code. One for normal path
+finding, one for aimless wandering (when you don't set orders) and one for longest path. This last
+one is a bit curious, and it appears to be used with placing down vehicles. I haven't quite got to
+the bottom of what it is there for yet.
+
+## Windows and UI
+
+@LeftOfZen has finally fixed the issue with scrollbars when you have a huge number of items. The
+scrollbar should no longer break.
+
+@ZehMatt has been fixing more of the fallout of adjusting how coordinates work, so hopefully now
+there aren't any graphical issues when you have the news window open.
+He also identified an issue with the landscape generation window resetting its state every tick,
+which presented itself in problems scrolling the terrain options.
+
+Last but not least, @LeeSpork has been adjusting the map window ensuring its minimum size is sensible.
+
+## Preparation for standalone
+
+As we are nearing the end of the reimplementation effort, we need to start getting ready for being standalone.
+This month, @ZehMatt has been replacing stored pointers in object code with memory offsets instead.
+This is a bit of trick that will simplify the code when we move to 64-bit.
```

Comment:
"This is a bit of trick..." sounds strange to me? Did you mean "...bit of a trick..."?